dask/backends.py

```diff
@@ -1,11 +1,17 @@
 from __future__ import annotations
 
 from functools import lru_cache, wraps
-from typing import Any, Callable, TypeVar, cast
+from typing import TYPE_CHECKING, Any, Callable, Generic, TypeVar
 
 from dask import config
 from dask.compatibility import entry_points
 
+if TYPE_CHECKING:
+    from typing_extensions import ParamSpec
+
+    BackendFuncParams = ParamSpec("BackendFuncParams")
+    BackendFuncReturn = TypeVar("BackendFuncReturn")
```

Comment:
Question for my own education: Does `TYPE_CHECKING` being `False` mean that the the `BackendFuncParams` and `BackendFuncReturn` usage in the `register_inplace` return-types will be ignored?

Reply:
At run-time they have no effect (at least in the way we are using types here). They really only mean anything to type checkers, when `TYPE_CHECKING` is `True`. When `TYPE_CHECKING` is `False`, these typevars are just opaque strings for the purposes of the python process parsing `register_inplace`.